db/hybrid_logical_vector.go

```diff
@@ -247,6 +273,12 @@ func (hlv *HybridLogicalVector) AddVersion(newVersion Version) error {
 		// source doesn't exist in PV so add
 		hlv.PreviousVersions[hlv.SourceID] = hlv.Version
 	}
+
+	// If new version already exists PV, need to remove it
+	if _, ok := hlv.PreviousVersions[newVersion.SourceID]; ok {
+		delete(hlv.PreviousVersions, newVersion.SourceID)
+	}
```

Comment:
take or leave, delete() when it doesn't exist is a no-op 
```suggestion
	delete(hlv.PreviousVersions, newVersion.SourceID)
```